skimage/future/graph/graph_merge.py

```diff
@@ -128,10 +134,16 @@ def merge_hierarchical(labels, rag, thresh, rag_copy, in_place_merge,
             merge_func(rag, src, dst)
             new_id = rag.merge_nodes(src, dst, weight_func)
             _revalidate_node_edges(rag, new_id, edge_heap)
+            if merge_trace:
+                label_map = np.arange(labels.max() + 1)
+                for ix, (n, d) in enumerate(rag.nodes_iter(data=True)):
+                    for label in d['labels']:
+                        label_map[label] = ix
+                trace.append(label_map[labels])
```

Comment:
@vighneshbirodkar yes, something along those lines is what I was thinking, but I think we have to make it a callable object. (In gala, I called these feature managers.) I don't see how just a simple function could work. But you are exactly right that almost all `weight_func` will need a very specific corresponding `describe_func` to work.